### DISCUSSION TITLE
Application error: a client-side exception has occurred (see the browser console for more information).

### DISCUSSION THREAD

**Discussion by ashikjhonson** [vercel/community#4576] [Help] [answered]
### Summary

Encountering a maximum stack error during the deployment of a Next.js app on Vercel.
I'm able to access main page of the site, but not on certain sub-routes.

<img width="352" alt="Screenshot 2023-10-23 150353" src="https://github.com/vercel/community/assets/97791612/069bfd3b-6cc6-4a6e-9b31-8b65db1d5175">

I think it's due to any mistakes while using any of the hooks, but I need some help for debugging this issue.
Thank you

### Example

https://ktu-cse-labs-site.vercel.app/semester/2/course/EST%20102%20PROGRAMMING%20IN%20C

### Steps to Reproduce

Just visit  https://ktu-cse-labs-site.vercel.app/semester/2
Click the button available there.

**Comment by krishnaacharyaa** [comment]
Hi @ashikjhonson, I guess you are trying to load all of your notes from the static files, you should probably  consider deploying it elsewhere and using the ref link load in the application, because nextjs cannot load all of your notes in the build as it exceeds its limit

**Reply by ashikjhonson** [reply]
Hey @krishnaacharyaa thanks for your support.
I can explain a bit more, I have uploaded my lab codes in my GitHub repo, now I'm building a site for others to access it easily.
The thing I'm doing here is like, I'm fetching the codes thru GitHub api, (only whenever a user accesses a specific course route).
Btw your reply for my issue seems to be relevant. I'll try to redesign the code.
Thanks dude 😊

**Reply by krishnaacharyaa** [reply, marked as answer]
Your welcome @ashikjhonson :) , 

When you say "only whenever a user accesses a specific course route", do you mean to say that you are dynamically calling the api ? , In case you didn't already know, nextjs by default if there is a scope to statically build api routes does it , unless explicitly specified. So it might accumulate large amount of files, which you might have to handle in specific.

**Reply by ashikjhonson** [reply]
Okay, now I'm thinking of replacing routes with object array elements instead of fetching dynamically  wherever possible.
